Assertion failed 'lastBlockILEndOffset < beginOffs'

Native AOT outerloop tests are failing on all architectures.

Example: https://dev.azure.com/dnceng-public/public/_build/results?buildId=613229&view=logs&jobId=0cd015bc-ac80-5dfb-09b8-6d9db63883bc&j=0cd015bc-ac80-5dfb-09b8-6d9db63883bc&t=3393f03a-e6e5-5ccb-ae92-d1ce7b161b0e

```
    ILC: D:\git\runtime2\src\coreclr\jit\scopeinfo.cpp:1572
    ILC: Assertion failed 'lastBlockILEndOffset < beginOffs' in 'ABIStress.Program:DoStubCall(int,ubyte,ubyte,int,int):
  ubyte' during 'Generate code' (IL size 2013; hash 0x4766f2b1; MinOpts)

D:\git\runtime2\artifacts\bin\coreclr\windows.x64.Checked\build\Microsoft.NETCore.Native.targets(310,5): error MSB3073:
 The command ""D:\git\runtime2\artifacts\bin\coreclr\windows.x64.Checked\ilc-published\\ilc" @"D:\git\runtime2\artifact
s\tests\coreclr/obj/windows.x64.Checked/Managed/JIT\Stress\ABI\tailcalls_d\native\tailcalls_d.ilc.rsp"" exited with cod
e -2147483645. [D:\git\runtime2\src\tests\JIT\Stress\ABI\tailcalls_d.csproj] [D:\git\runtime2\src\tests\build.proj]
```

JitDump: 
[jitdump.zip](https://github.com/dotnet/runtime/files/14718946/jitdump.zip)

--- a/src/coreclr/jit/helperexpansion.cpp
+++ b/src/coreclr/jit/helperexpansion.cpp
@@ -1504,15 +1504,8 @@ bool Compiler::fgExpandStaticInitForCall(BasicBlock** pBlock, Statement* stmt, G
     // Clear gtInitClsHnd as a mark that we've already visited this call
     call->gtInitClsHnd = NO_CLASS_HANDLE;
 
-    // Update the IL offsets of the blocks to match the split.
-
-    // prevBb->bbCodeOffs remains the same
-    block->bbCodeOffsEnd = prevBb->bbCodeOffsEnd;
-
-    IL_OFFSET splitPointILOffset = fgFindBlockILOffset(block);
-
-    prevBb->bbCodeOffsEnd = max(prevBb->bbCodeOffs, splitPointILOffset);
-    block->bbCodeOffs     = min(splitPointILOffset, block->bbCodeOffsEnd);
+    // The blocks and statements have been modified enough to make the ending offset invalid.
+    block->bbCodeOffsEnd = BAD_IL_OFFSET;
 
     return true;
 }

--- a/src/tests/JIT/Stress/ABI/pinvokes_d.csproj
+++ b/src/tests/JIT/Stress/ABI/pinvokes_d.csproj
@@ -10,9 +10,6 @@
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <JitOptimizationSensitive>true</JitOptimizationSensitive>
     <GCStressIncompatible>true</GCStressIncompatible>
-
-    <!-- https://github.com/dotnet/runtime/issues/100125 -->
-    <NativeAotIncompatible>true</NativeAotIncompatible>
   </PropertyGroup>
   <PropertyGroup>
     <DebugType>Full</DebugType>

--- a/src/tests/JIT/Stress/ABI/pinvokes_do.csproj
+++ b/src/tests/JIT/Stress/ABI/pinvokes_do.csproj
@@ -10,9 +10,6 @@
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <JitOptimizationSensitive>true</JitOptimizationSensitive>
     <GCStressIncompatible>true</GCStressIncompatible>
-
-    <!-- https://github.com/dotnet/runtime/issues/100125 -->
-    <NativeAotIncompatible>true</NativeAotIncompatible>
   </PropertyGroup>
   <PropertyGroup>
     <DebugType>Full</DebugType>

--- a/src/tests/JIT/Stress/ABI/stubs_do.csproj
+++ b/src/tests/JIT/Stress/ABI/stubs_do.csproj
@@ -10,9 +10,6 @@
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <JitOptimizationSensitive>true</JitOptimizationSensitive>
     <GCStressIncompatible>true</GCStressIncompatible>
-
-    <!-- https://github.com/dotnet/runtime/issues/100125 -->
-    <NativeAotIncompatible>true</NativeAotIncompatible>
   </PropertyGroup>
   <PropertyGroup>
     <DebugType>Full</DebugType>

--- a/src/tests/JIT/Stress/ABI/tailcalls_d.csproj
+++ b/src/tests/JIT/Stress/ABI/tailcalls_d.csproj
@@ -10,9 +10,6 @@
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <JitOptimizationSensitive>true</JitOptimizationSensitive>
     <GCStressIncompatible>true</GCStressIncompatible>
-
-    <!-- https://github.com/dotnet/runtime/issues/100125 -->
-    <NativeAotIncompatible>true</NativeAotIncompatible>
   </PropertyGroup>
   <PropertyGroup>
     <DebugType>Full</DebugType>

--- a/src/tests/JIT/Stress/ABI/tailcalls_do.csproj
+++ b/src/tests/JIT/Stress/ABI/tailcalls_do.csproj
@@ -10,9 +10,6 @@
     <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
     <JitOptimizationSensitive>true</JitOptimizationSensitive>
     <GCStressIncompatible>true</GCStressIncompatible>
-
-    <!-- https://github.com/dotnet/runtime/issues/100125 -->
-    <NativeAotIncompatible>true</NativeAotIncompatible>
   </PropertyGroup>
   <PropertyGroup>
     <DebugType>Full</DebugType>

